### PR TITLE
PAPP-28270: Elasticsearch - Update the API to the latest version

### PR DESCRIPTION
**Pull request by aparekh-crest**
…hooks

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: readme.html [hunk 1]
```diff
@@ -13,4 +13,19 @@
 and limitations under the License.
 -->
 
-<p>Elasticsearch installations can be configured to allow REST API access without any type of authentication. The app therefore marks <b>username</b> and <b>password</b> as optional parameters. If specified, the app will end up using these for generating the basic authentication header used in the various Elasticsearch REST endpoints.<br>Connection can be configured over HTTP or HTTPS, so if <b>test connectivity</b> fails please check the protocol.</p>
+<p>Elasticsearch installations can be configured to allow REST API access without any type of authentication.
```
Comment (splunk-jessica) [inline readme.html:16]:
```suggestion
<p>Elasticsearch installations can be configured to allow REST API access without any type of authentication.
    The app therefore marks <b>username</b> and <b>password</b> as optional parameters. If specified, the app
    will use these for generating the basic authentication header for various Elasticsearch
    REST endpoints.<br>The connection can be configured over HTTP or HTTPS, so if <b>test connectivity</b>
    fails please check the protocol.
</p>
```

File: readme.html [hunk 2]
```diff
@@ -13,4 +13,19 @@
...
+    The app therefore marks <b>username</b> and <b>password</b> as optional parameters. If specified, the app
+    will end up using these for generating the basic authentication header used in the various Elasticsearch
```
Comment (splunk-jessica) [inline readme.html:18]:
```suggestion
    will use these for generating the basic authentication header for various Elasticsearch
```

File: readme.html [hunk 3]
```diff
@@ -13,4 +13,19 @@
...
+    REST endpoints.<br>Connection can be configured over HTTP or HTTPS, so if <b>test connectivity</b>
```
Comment (splunk-jessica) [inline readme.html:19]: REST endpoints.<br>The connection can be configured over HTTP or HTTPS, so if <b>test connectivity</b>

File: readme.html [hunk 4]
```diff
@@ -13,4 +13,19 @@
...
+    fails please check the protocol.
+</p>
+<p><b>Playbook Backward Compatibility</b><br>
+    Action parameter has been removed to the action given below. Hence, It is requested to the end-user please update their existing playbooks parameters to ensure the correct functioning
```
Comment (splunk-jessica) [inline readme.html:23]:
```suggestion
    An action parameter has been removed from the run query action. Hence, it is recommended for the end-user to update their existing playbooks parameters to ensure that playbooks created on earlier versions of the app are functioning correctly
```

File: readme.html [hunk 5]
```diff
@@ -13,4 +13,19 @@
...
+    of the playbooks created on the earlier versions of the app.
```
Comment (splunk-jessica) [inline readme.html:24]:
```suggestion
The following action parameter has been changed as detailed below. Hence, it is recommended that the end-user update their existing playbooks parameters accordingly to ensure that playbooks created using earlier versions of the app are functioning correctly.
```

File: elasticsearch.json [hunk 6]
```diff
@@ -152,13 +173,13 @@
...
+            "verbose": "The action executes the query on an Elasticsearch installation by doing a POST on the REST endpoint '<b>base_url</b>/<b>index</b>/_search' with the input <b>query</b> as the data, if specified. Please see the Elasticseach website for query format and documentation.<br>The <b>routing</b> parameter is appended as a parameter in the REST call if specified.<br>As an e.g. the following query returns only the <i>id</i> and <i>name</i> of all the items from the given <b>index</b><br>{ \"query\": { \"match_all\": {} }, \"_source\": [\"id\", \"name\"]}.",
             "type": "investigate",
             "identifier": "run_query",
             "read_only": true,
             "parameters": {
                 "index": {
-                    "description": "Index to query on",
+                    "description": "Comma seperated list of indexes to query on",
```
Comment (splunk-jessica) [inline elasticsearch.json:182]:
```suggestion
                    "description": "Comma-separated list of indexes to query on",
```

File: readme.html [hunk 7]
```diff
@@ -13,4 +13,19 @@
...
+    <ul>
+        <li><b>For version 2.1.0 :</b></li>
```
Comment (splunk-jessica) [inline readme.html:26]: Is the "version 2.1.0" referencing the last version of the Elasticsearch connector that was released? According the previous release notes, the last released version was 2.0.5 not 2.1.0. I would suggest something like this, assuming the new release will be v3.0.0 (as defined in the JSON file):

```suggestion
        <li><b>Changes from version 2.0.5 to version 3.0.0:</b></li>
```

**Review by tonyc-phantom — APPROVED** [review]

File: readme.html (hunk 7 above)
Comment (aparekh-crest) [reply readme.html]: Yes it should be v3.0.0, updated the changes

File: elasticsearch.json (hunk 6 above)
Comment (aparekh-crest) [reply elasticsearch.json]: Done with the changes

**Pull request merged**